winch/codegen/src/isa/x64/masm.rs

```diff
@@ -418,6 +418,57 @@ impl Masm for MacroAssembler {
     fn jmp(&mut self, target: MachLabel) {
         self.asm.jmp(target);
     }
+
+    fn popcnt(&mut self, src: Reg, dst: Reg, size: OperandSize) {
+        if self.flags.has_popcnt() {
+            self.asm.popcnt(src, dst, size)
+        } else {
+            let (masks, shift_amt) = match size {
+                OperandSize::S64 => (
+                    [
+                        0x5555555555555555, // m1
+                        0x3333333333333333, // m2
+                        0x0f0f0f0f0f0f0f0f, // m4
+                        0x0101010101010101, // h01
+                    ],
+                    56u8,
+                ),
+                // 32-bit popcount is the same, except the masks are half as
+                // wide and we shift by 24 at the end rather than 56
+                OperandSize::S32 => (
+                    [0x55555555i64, 0x33333333i64, 0x0f0f0f0fi64, 0x01010101i64],
+                    24u8,
+                ),
+            };
+            let tmp = regs::scratch();
+            self.asm.mov_rr(src, tmp, size);
+            if src != dst {
+                self.asm.mov_rr(src, dst, size);
+            }
+
+            // x -= (x >> 1) & m1;
+            self.asm.shift_ir(1u8, dst, ShiftKind::ShrU, size);
+            self.asm.and(RegImm::imm(masks[0]).into(), dst.into(), size);
+            self.asm.sub(dst.into(), tmp.into(), size);
+
+            // x = (x & m2) + ((x >> 2) & m2);
+            self.asm.mov(tmp.into(), dst.into(), size);
```

Comment:
Can this and the other `mov` below be written like this? I'm not sure it makes any difference but it seems a little more clear to me if we can avoid using `.into()` so much. I'm also curious if there are other methods named perhaps `sub_rr` or `and_ir` for the other instructions in this sequence.
```suggestion
            self.asm.mov_rr(tmp, dst, size);
```